client/my-sites/plans/jetpack-plans/iterations.ts

```diff
@@ -19,21 +11,77 @@ export enum Iterations {
 	I5 = 'i5',
 	SPP = 'spp',
 }
+const iterationNames: string[] = Object.values( Iterations );
 
 /**
  * Getters
  */
 
-export function getGridComponent(): React.FC< ProductsGridProps > | undefined {
-	return {
-		[ Iterations.I5 ]: ProductsGridI5,
-		[ Iterations.SPP ]: ProductsGridSpp,
-	}[ getIteration() as Iterations ];
-}
+/**
+ * Gets the name of the current CRO iteration.
+ *
+ * **NOTE:** Avoid using this if you can; instead, opt for either `getForCurrentCROIteration` or `doForCurrentCROIteration`.
```

Comment:
Out of curiosity, why should we avoid using this function and opt for one of the others? It seems to me that all of them use the `abtest` function.

If calling `abtest` is what we want to avoid, couldn't we cache the value it returns and on subsequent calls return the cached value instead? I could be wrong, but I'm under the impression that changing the test branch triggers a page reload so the cached value won't ever be wrong.

Reply:
In earlier iterations I was trying to 100% avoid exposing the current iteration name anywhere outside of `iterations.ts`. I backed off after seeing that we use the literal iteration name for styling in two places:

* `client/jetpack-cloud/sections/pricing/header/index.tsx`
* `client/jetpack-cloud/sections/partner-portal/header/index.tsx`

If possible I think it would be a good move for us to limit visibility of the current iteration name as much as we can, to avoid situations where we have components with lots of hard-coded iteration name strings. In the end though, it's mostly a stylistic choice, so I understand if it comes off a bit silly. 🙂

Reply:
Oh, I thought all of this was to avoid calling `abtest( 'jetpackSimplifyPricingPage' )` multiple times because it does some side-effect we didn't want to happen, that's why I suggested caching the value it returns.

Now I understand that it's about hiding the current iteration which sounds very smart to me. Thanks for explaining it!

Reply:
I just made a change to remove the `export` from `getCurrentCROIterationName`. Turns out it was easy enough to change the affected style rules and have them use the getter like everything else.